File: src/main/java/ru/mm/surv/config/HttpTraceActuatorConfiguration.java
package ru.mm.surv.config;

import org.springframework.boot.actuate.audit.AuditEventRepository;
import org.springframework.boot.actuate.audit.InMemoryAuditEventRepository;
import org.springframework.boot.actuate.trace.http.HttpExchangeTracer;
import org.springframework.boot.actuate.trace.http.HttpTraceRepository;
import org.springframework.boot.actuate.trace.http.InMemoryHttpTraceRepository;
import org.springframework.boot.actuate.web.trace.servlet.HttpTraceFilter;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

@Configuration
public class HttpTraceActuatorConfiguration {

    private static final int ORDER_BEFORE_SECURITY_FILTER = -102;

    @Bean
    public AuditEventRepository auditEventRepository() {
        return new InMemoryAuditEventRepository();
    }

    @Bean
    public HttpTraceFilter httpTraceFilter(HttpTraceRepository repository, HttpExchangeTracer tracer) {
        var HttpTraceFilter = new HttpTraceFilterWithPrincipal(repository, tracer);
        HttpTraceFilter.setOrder(ORDER_BEFORE_SECURITY_FILTER);
        return HttpTraceFilter;
    }

    @Bean
    public HttpTraceRepository httpTraceRepository() {
        return new InMemoryHttpTraceRepository();
    }
}

File: src/main/java/ru/mm/surv/config/HttpTraceActuatorConfiguration.kt
package ru.mm.surv.config

import org.springframework.boot.actuate.audit.AuditEventRepository
import org.springframework.boot.actuate.audit.InMemoryAuditEventRepository
import org.springframework.boot.actuate.trace.http.HttpTraceRepository
import org.springframework.boot.actuate.trace.http.HttpExchangeTracer
import org.springframework.boot.actuate.web.trace.servlet.HttpTraceFilter
import org.springframework.boot.actuate.trace.http.InMemoryHttpTraceRepository
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

private const val ORDER_BEFORE_SECURITY_FILTER = -102

@Configuration
class HttpTraceActuatorConfiguration {

    @Bean
    fun auditEventRepository(): AuditEventRepository {
        return InMemoryAuditEventRepository()
    }

    @Bean
    fun httpTraceFilter(repository: HttpTraceRepository, tracer: HttpExchangeTracer): HttpTraceFilter {
        val httpTraceFilter = HttpTraceFilterWithPrincipal(repository, tracer)
        httpTraceFilter.order = ORDER_BEFORE_SECURITY_FILTER
        return httpTraceFilter
    }

    @Bean
    fun httpTraceRepository(): HttpTraceRepository {
        return InMemoryHttpTraceRepository()
    }
}
